diff --git a/sgsr/v1/include/glsl/sgsr1_shader_mobile.frag b/sgsr/v1/include/glsl/sgsr1_shader_mobile.frag
index 2db04b3..056d177 100644
--- a/sgsr/v1/include/glsl/sgsr1_shader_mobile.frag
+++ b/sgsr/v1/include/glsl/sgsr1_shader_mobile.frag
@@ -8,8 +8,10 @@
 //
 //============================================================================================================
 
-precision mediump float;
-precision highp int;
+//precision mediump float;
+//precision highp int;
+#define highp
+#define mediump
 
 ////////////////////////
 // USER CONFIGURATION //
diff --git a/sgsr/v1/include/glsl/sgsr1_shader_mobile_edge_direction.frag b/sgsr/v1/include/glsl/sgsr1_shader_mobile_edge_direction.frag
index 5a38589..5ed5f5e 100644
--- a/sgsr/v1/include/glsl/sgsr1_shader_mobile_edge_direction.frag
+++ b/sgsr/v1/include/glsl/sgsr1_shader_mobile_edge_direction.frag
@@ -8,8 +8,10 @@
 //
 //============================================================================================================
 
-precision mediump float;
-precision highp int;
+//precision mediump float;
+//precision highp int;
+#define highp
+#define mediump
 
 ////////////////////////
 // USER CONFIGURATION //
