diff --git a/include/util/util_small_vector.h b/include/util/util_small_vector.h
--- a/include/util/util_small_vector.h
+++ b/include/util/util_small_vector.h
@@ -193,7 +193,10 @@

     size_t pick_capacity(size_t n) {
       // Pick next largest power of two for the new capacity
-      return size_t(1u) << ((sizeof(n) * 8u) - bit::lzcnt(n - 1));
+      if constexpr (sizeof(size_t) == sizeof(uint32_t))
+        return size_t(1u) << ((sizeof(n) * 8u) - bit::lzcnt(uint32_t(n - 1)));
+      else
+        return size_t(1u) << ((sizeof(n) * 8u) - bit::lzcnt(uint64_t(n - 1)));
     }

     T* ptr(size_t idx) {
