--- a/include/dxbc/dxbc_util.h
+++ b/include/dxbc/dxbc_util.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <array>
+
 #include "dxbc_common.h"
 #include "dxbc_enums.h"
 
--- a/include/dxbc/dxbc_compiler.h
+++ b/include/dxbc/dxbc_compiler.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <array>
+#include <initializer_list>
 #include <utility>
 #include <vector>
 
--- a/include/util/util_bit.h
+++ b/include/util/util_bit.h
@@ -1,5 +1,8 @@
 #pragma once
 
+#include <algorithm>
+#include <limits>
+
 #if (defined(__x86_64__) && !defined(__arm64ec__)) || (defined(_M_X64) && !defined(_M_ARM64EC)) \
     || defined(__i386__) || defined(_M_IX86) || defined(__e2k__)
   #define DXVK_ARCH_X86
--- a/include/util/util_string.h
+++ b/include/util/util_string.h
@@ -2,6 +2,7 @@
 
 #include <cstring>
 #include <string>
+#include <string_view>
 #include <sstream>
 #include <vector>
 
--- a/include/util/util_error.h
+++ b/include/util/util_error.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <string>
+#include <utility>
 
 namespace dxvk {
   
--- a/src/dxbc/dxbc_defs.cpp
+++ b/src/dxbc/dxbc_defs.cpp
@@ -1,3 +1,5 @@
+#include <array>
+
 #include "dxbc_defs.h"
 
 namespace dxvk {
--- a/src/dxbc/dxbc_chunk_isgn.cpp
+++ b/src/dxbc/dxbc_chunk_isgn.cpp
@@ -1,3 +1,5 @@
+#include <algorithm>
+
 #include "dxbc_chunk_isgn.h"
 
 namespace dxvk {
--- a/src/dxbc/dxbc_compiler.cpp
+++ b/src/dxbc/dxbc_compiler.cpp
@@ -1,3 +1,5 @@
+#include <algorithm>
+
 #include "dxbc_compiler.h"
 
 namespace dxvk {
--- a/src/spirv/spirv_module.cpp
+++ b/src/spirv/spirv_module.cpp
@@ -1,3 +1,4 @@
+#include <array>
 #include <cstring>
 
 #include "spirv_module.h"
--- a/src/spirv/spirv_code_buffer.cpp
+++ b/src/spirv/spirv_code_buffer.cpp
@@ -1,5 +1,6 @@
 #include <array>
 #include <cstring>
+#include <limits>
 
 #include "spirv_code_buffer.h"
 
